.github/workflows/retrieve-commits.yml

```
@@ -0,0 +1,39 @@
+
+      - name: Get specific commit
+        run: |
+          gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            /repos/${{ github.repository_owner }}/${{ github.event.repository.name }}/commits/${{ github.sha }} \
+            > commit.json
+
+      
+      - name: Upload commit data
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)